cloud/scope/managed_machine_pool.go

```diff
@@ -199,6 +199,15 @@ func (m *ManagedMachinePoolScope) FindNodePool(ctx context.Context) (*oke.NodePo
 	return nil, nil
 }
 
+func (m *ManagedMachinePoolScope) getNodePoolName() string {
+	nodePoolName := m.OCIManagedMachinePool.GetName()
+	// oke nopdepool names are restricted to less than 32
+	if len(nodePoolName) > 31 {
+		return nodePoolName[0:31]
```

Comment:
what about adding this to the webhook do this? or maybe have a single `handleLongOKEName` or some other single use util function? I feel like we might forget this check is being done in two places. Or at least I will probably forget 😆

Reply:
same as above